### PR TITLE
Use Static ClientLoggers in azure-search-documents

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Updates `azure-search-documents` to use static `ClientLogger`s.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ScoringParameter.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
  * Represents a parameter value to be used in scoring functions (for example, referencePointParameter).
  */
 public final class ScoringParameter {
-    private final ClientLogger logger = new ClientLogger(ScoringParameter.class);
+    private static final ClientLogger LOGGER = new ClientLogger(ScoringParameter.class);
```
Comment (pallavit) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ScoringParameter.java:24]: Is this field not going on the wire? It does not have JsonIgnore on it?

The other two model types SearchResults, SuggestResult seem to be responses so I think those don't need it.

**Review by pallavit — APPROVED** [review]
:shipit:

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ScoringParameter.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ScoringParameter.java:24]: This goes over the wire but is handled manually as it needs to have a specific conversion applied

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java [hunk 2]
```diff
@@ -681,7 +678,7 @@ public String getEndpoint() {
                         try {
                             return new SimpleResponse<>(res, Utility.convertValue(res.getValue(), modelClass));
                         } catch (IOException ex) {
-                            throw logger.logExceptionAsError(
+                            throw LOGGER.logExceptionAsError(
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java:681]: Unrelated to this PR - but we should avoid directly throwing RuntimeException if a more specific exception can be thrown. In this case, we could use `UncheckedIOException`.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/batching/SearchIndexingPublisher.java [hunk 3]
```diff
@@ -143,11 +143,11 @@ public synchronized Duration getCurrentRetryDelay() {
                 this.actions.add(action);
             });
 
-        logger.verbose("Actions added, new pending queue size: {}.", this.actions.size());
+        LOGGER.verbose("Actions added, new pending queue size: {}.", this.actions.size());
 
         if (autoFlush && batchAvailableForProcessing()) {
             rescheduleFlush.run();
-            logger.verbose("Adding documents triggered batch size limit, sending documents for indexing.");
+            LOGGER.verbose("Adding documents triggered batch size limit, sending documents for indexing.");
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/batching/SearchIndexingPublisher.java:150]: Again unrelated to this change but `verbose` logs should use the supplier API in ClientLogger.

**Review by srnagar — APPROVED** [review]

**Pull request merged**